frontend/src/utilities/index.tsx

```diff
@@ -19,3 +21,45 @@ export function handleAutoResize(textarea: HTMLTextAreaElement): void {
 	textarea.style.height = '100%';
 	textarea.style.height = `${textarea.scrollHeight}px`;
 }
+
+/**
+ * Hook to manage the document context in the editor.
+ */
+export function useDocContext(editorAPI: EditorAPI) {
```

Comment:
I kind of imagine this as being a Context provided by the editor API, so we could `useContext` it without having to pass `editorAPI` around. But I'm not certain of that design.